priv/repo/migrations/20240626103903_create_activity_logs.exs

```diff
@@ -0,0 +1,18 @@
+defmodule Trento.Repo.Migrations.CreateActivityLogs do
+  use Ecto.Migration
+
+  def change do
+    create table(:activity_logs, primary_key: false) do
+      add :id, :uuid, primary_key: true
+      add :type, :string
+      add :actor, :string
+      add :metadata, :map
+
+      timestamps(type: :utc_datetime_usec)
+    end
+
+    create index(:activity_logs, [:type])
+    create index(:activity_logs, [:actor])
+    create index(:activity_logs, [:inserted_at])
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏What if we create indexes later? I'm a bit concerned about the improvements vs the added drawbacks. Should we do some performance profiles before tuning the db with indexes?

Maybe it's just me overthinking, though.

_Rationale_
Being logging a _write-intensive_ operation, I'm afraid of the negative impact on performance to the traced activities (that is: the faster the write, the better). Rule of thumb, I think we might afford more a _very slow_ read once in a while than a _slightly slow_ write many times.

Reply:
sure, it makes sense. Let's remove indexes for now

Reply:
ae9032c86efa85b6b110a9bf1a6266f876c9cfd8